app/services/job_tracker.rb

```diff
@@ -0,0 +1,45 @@
+class JobTracker
+  VALID_STATE_KEYS = [:message, :state].freeze
+  DAY_IN_SECONDS = 86_400
+
+  attr_accessor :job_id, :queue_name
+
+  def initialize(job_id:, queue_name:)
+    raise 'Missing job identifiers!' unless job_id && queue_name
+
+    @job_id = job_id
+    @queue_name = queue_name
+
+    redis.expire(redis_key, DAY_IN_SECONDS)
+  end
+
+  def get_state
+    state_hash = JSON.parse(redis.get(redis_key) || '{}')
+    state_hash.symbolize_keys
+  end
+
+  def set_state(state_hash)
+    return unless valid_keys?(state_hash)
+
+    redis.set(redis_key, state_hash.to_json, keepttl: true)
+  end
+
+  def update_state(state)
+    state_hash = get_state
+    set_state(state_hash.merge(state))
```

Comment:
can probably combine in a single method checking if it exists? Not sure how common is set vs update